src/weapon/w_007.c

```diff
@@ -67,11 +239,100 @@ void EntityWeaponAttack(Entity* self) {
     self->rotPivotY = PLAYER.rotPivotY;
 }
 
-INCLUDE_ASM("weapon/nonmatchings/w_007", func_ptr_80170004);
-
-extern u8 D_35000_8017AC64[6][8];
+static void func_ptr_80170004(Entity* self) {
+    Entity* parent;
+    s32 temp_v1_3;
+    s32 temp_v1_4;
+    u16 a0;
+    u16 v0;
+
+    switch (self->step) {
+    case 0:
+        parent = self->ext.weapon.parent;
+        if (parent->entityId != 0) {
+            temp_v1_3 = (self->params >> 7);
+            self->facingLeft = parent->facingLeft;
+            a0 = D_35000_8017AC48[temp_v1_3];
+            v0 = D_35000_8017AC48[temp_v1_3 + 1];
+            if (self->facingLeft) {
+                a0 = -a0;
+            }
+            self->posX.i.hi += a0;
+            self->posY.i.hi += v0;
+
+            self->animSet = self->ext.weapon.parent->animSet;
+            self->anim = D_35000_8017AB44;
+            self->zPriority = 0xA8;
+            self->unk5A = self->ext.weapon.parent->unk5A;
+            self->palette = self->ext.weapon.parent->palette;
+
+            SetSpeedX(FIX(4.75));
+            temp_v1_4 = self->params & 0xFF00;
+            if ((temp_v1_4 == 0x300) || (temp_v1_4 == 0x600)) {
```

Comment:
Looks like temp_v1_4 is not a real variable, I tried removing it in the scratch and still matched.